src/pgroonga.c

```diff
@@ -1170,6 +1171,44 @@ PGrnIsForRegexpSearchIndex(Relation index, int nthAttribute)
 	return OidIsValid(regexpStrategyOID);
 }
 
+static bool
+PGrnIsRegexpStrategyV2Index(Relation index, int nthAttribute)
+{
+	Oid regexpStrategyOID;
+	Oid leftType;
+	Oid rightType;
+
+	leftType = index->rd_opcintype[nthAttribute];
+
+	switch (leftType)
+	{
+	case TEXTARRAYOID:
+		rightType = TEXTOID;
+		break;
+	default:
+		rightType = leftType;
+		break;
+	}
+
+	regexpStrategyOID = get_opfamily_member(index->rd_opfamily[nthAttribute],
+											leftType,
+											rightType,
+											PGrnRegexpStrategyV2Number);
+	return OidIsValid(regexpStrategyOID);
+}
+
+static bool
+PGrnIsForRegexpSearchIndex(Relation index, int nthAttribute)
+{
+	if (PGrnIsRegexpStrategyIndex(index, nthAttribute))
+		return true;
+
+	if (PGrnIsRegexpStrategyV2Index(index, nthAttribute))
+		return true;
+
+	return false;
```

Comment:
How about writing it like the following because both of them return bool type.
```suggestion
	return PGrnIsRegexpStrategyIndex(index, nthAttribute) ||
               PGrnIsRegexpStrategyV2Index(index, nthAttribute);
```

Reply:
既存の類似関数がこのスタイルなのでこのままでいいっす。